pkg/schemas/secret.go

```diff
@@ -0,0 +1,84 @@
+package schemas
+
+import (
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// DescribeSecretSchema represents output of DESCRIBE query for the single secret.
+var DescribeSecretSchema = map[string]*schema.Schema{
+	"created_on": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"name": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"database_name": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"schema_name": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"owner": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"comment": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"secret_type": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"username": {
+		Type:     schema.TypeString,
+		Optional: true,
```

Comment:
Describe for secret in Snowflake leaves some of those fields as "null" since secret can be one of four different types. Thought it needs to be included here as well, but it works as it should after removing the Optional from those fields.